### ISSUE TITLE
C#: [Tool] Export Node: setter is not called from editor inspector

### ISSUE THREAD

**Issue by rndevfx** [godotengine/godot#77685]
### Godot version

4.0.3.stable

### System information

MacOS Ventura 13.3.1 (a)

### Issue description

I want to set the shape radius of a `CollisionShape3D` from the editor via script. So I exported a new `CollisionShape3D` but the setter is never called when I select a node from the inspector. So I cannot access the node within the script.
```cs
private CollisionShape3D _collisionShape3D;

[Export]
public CollisionShape3D CollisionShape3D
{
    get => _collisionShape3D;
    set
    {
        _collisionShape3D = value;
        GD.Print("Setter called for CollisionShape3D");
    }
}
```

When I do the same with an `int` it works fine.

### Steps to reproduce

Create a new node, add the C# snippet to the Node, and try to select another Node out of the inspector. The setter for the `CollisionShape3D` is never called from the editor. It's called only when the project is started.

```cs
using Godot;

[Tool]
public partial class Node3D : Godot.Node3D
{
    private int _dummyInt;
    private CollisionShape3D _collisionShape3D;

    [Export]
    public int DummyInt
    {
        get => _dummyInt;
        set
        {
            _dummyInt = value;
            GD.Print("Setter called for int"); // This works fine from editor inspector
        }
    }

    [Export]
    public CollisionShape3D CollisionShape3D
    {
        get => _collisionShape3D;
        set
        {
            _collisionShape3D = value;
            GD.Print("Setter called for CollisionShape3D"); // Not called from editor inspector
        }
    }
}
```

### Minimal reproduction project

[GodotCSSetterIssue.zip](https://github.com/godotengine/godot/files/11609802/GodotCSSetterIssue.zip)

**Comment by AThousandShips** [comment]
Yes most likely a duplicate of:
* #74050

**Comment by Calinou** [comment]
- Duplicate of https://github.com/godotengine/godot/issues/74050, which is fixed in 4.1.
